### ISSUE TITLE
Re-running the same data produces different results?

### ISSUE THREAD

**Issue by River366** [AntonioDeFalco/SCEVAN#36]
Dear Mr. AntonioDeFalco,
Last day, I just re-running one sample data, then I found it seems like the results have some different. The tsne_scRNA.png, subclone numbers and tumor-No tumor identify was same. But the DEchr gene volcano figure were all different, maybe due to the subclone order changed?  First run, these volcano figures come from chr(1,4,6,7,12,14), however, in 2nd run it only have chr(4,7,14).
So, I feeling some confused about that,  what the meaning for those DEchr gene volcano figures?

Thanks for your help!
![1st](https://user-images.githubusercontent.com/111686066/192673782-3eba8dfd-3bd8-4702-adcb-eacde2fb3120.png)
![2nd](https://user-images.githubusercontent.com/111686066/192673798-37551cce-5cb4-42dd-ae29-a64d85db3c57.png)

**Comment by River366** [comment]
Sorry for my trouble. In fact, in some other samples, the subclone numbers also changed. That make the results mostly different. Would you mind provide some help?

**Comment by AntonioDeFalco** [comment]
Strange that you get different results, I repeated the analysis several times, it never happened before. I see that the sample name is NS_19 is by chance the sample of this dataset https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131907. I ask this in order to try to understand the problem using the same matrix. 

Thanks for your interest. Regards.

**Comment by River366** [comment]
Yes,the data just from GSE131907. If you have time, please try NS_02, NS_07, and NS_19 which have some different results. The other point is, does those DEchr volcano figures meaning for difference gene expression in regions of chromosomal variation?  If there are overlapping regions, then the result looks a little strange(LX255B from GSE123902). 

Thanks for your help! look forward to hearing from you soon.
Best regards.

![20220929175121](https://user-images.githubusercontent.com/111686066/192987133-b07445a0-e8e4-4691-886c-99ff7831cd23.png)

**Comment by dgacquer** [comment]
Any update on this ?

I am having the same issue. If I run pipelineCNA 10 times with the same data as input, I will get 3 different results randomly. 

The tsne_CNA plot is always the same, but the clustering membership (dot color) differs between runs. I have tried to modify the source code by adding a set.seed(123) after  library(igraph) in the subclonesTumorCells function but it did not solve the problem.

Best regards,

David
